src/cascade/input_data/db/csmr.py

```diff
@@ -0,0 +1,136 @@
+"""Upload csmr data to t3 table so EpiViz can access it for plotting."""
+
+import logging
+
+try:
+    from db_queries import get_outputs
+except ImportError:
+
+    class DummyGetOutputs:
+        def __getattr__(self, name):
+            raise ImportError(f"Required package db_queries not found")
+
+    get_outputs = DummyGetOutputs()
+
+
+from cascade.core.db import cursor
+
+
+CODELOG = logging.getLogger(__name__)
+
+
+def _csmr_in_t3(execution_context):
+    """Checks if data for the current model_version_id already exists in tier 3.
+    """
+
+    model_version_id = execution_context.parameters.model_version_id
+
+    query = """
+    SELECT exists(
+             SELECT * FROM epi.t3_model_version_csmr
+             WHERE model_version_id = %(model_version_id)s
+    )
+    """
+    with cursor(execution_context) as c:
+        c.execute(query, args={"model_version_id": model_version_id})
+        exists = c.fetchone()[0]
+
+    return exists == 1
+
+
+def _get_csmr_data(execution_context):
+
+    model_version_id = execution_context.parameters.model_version_id
+    add_csmr_cause = execution_context.parameters.add_csmr_cause
+    parent_loc = execution_context.parameters.drill
```

Comment:
Why `drill`? I would call this `location_id` or `node_id`. The context will (I think) only ever be in reference to one node at a time.

Reply:
ok.  I was thinking of the JSON settings string.

Reply:
Yes, the drill, for the example we've seen, did contain the parent. I'm guessing there will be a translation step from the settings to a more generic parent-child tree. The execution context might contain both parent and children, or it could have a view of the tree, if that's what we need. Unclear now.